fili-core/src/test/groovy/com/yahoo/bard/webservice/data/config/table/BaseTableLoaderSpec.groovy

```diff
@@ -8,95 +8,269 @@ import com.yahoo.bard.webservice.data.config.ResourceDictionaries
 
-        public SimpleBaseTableLoader(DataSourceMetadataService metadataService) {
+        SimpleBaseTableLoader(DataSourceMetadataService metadataService) {
             super(metadataService)
         }
 
         @Override
-        public void loadTableDictionary(ResourceDictionaries dictionaries) {
+        void loadTableDictionary(ResourceDictionaries dictionaries) {
             // stub to allow testing of abstract class in unit tests
         }
     }
 
+    private static class SimpleDependencyPhysicalTableDefinition extends PhysicalTableDefinition {
+        String dependentTableName
+        PhysicalTable physicalTable
+
+        SimpleDependencyPhysicalTableDefinition(String tableName, String dependentTableName) {
+            super(TableName.of(tableName), DefaultTimeGrain.DAY.buildZonedTimeGrain(DateTimeZone.UTC), [] as Set, [] as Set)
+            this.dependentTableName = dependentTableName
+        }
+
+        SimpleDependencyPhysicalTableDefinition(String tableName, PhysicalTable physicalTable) {
+            super(TableName.of(tableName), DefaultTimeGrain.DAY.buildZonedTimeGrain(DateTimeZone.UTC), [] as Set, [] as Set)
+            this.physicalTable = physicalTable
+        }
+
+        @Override
+        Set<TableName> getDependentTableNames() {
+            if (Objects.isNull(dependentTableName)) {
+                return Collections.emptySet()
+            }
+            return Collections.singleton(TableName.of(dependentTableName));
+        }
+
+        @Override
+        Optional<PhysicalTable> build(
+                ResourceDictionaries dictionaries,
+                DataSourceMetadataService metadataService
+        ) {
+            if (!Objects.isNull(physicalTable)) {
+                return Optional.of(physicalTable)
+            }
+            return dictionaries.getPhysicalDictionary().containsKey(dependentTableName) ?
+                    Optional.of(
+                            new ConcretePhysicalTable(
+                                    TableName.of(getName().asName()),
+                                    DefaultTimeGrain.DAY.buildZonedTimeGrain(DateTimeZone.UTC),
+                                    [] as Set,
+                                    [:],
+                                    metadataService
+                            )
+                    ) : Optional.empty()
+        }
+    }
+
     BaseTableLoader loader
     ResourceDictionaries dicts
     Set<ApiMetricName> apiNames
     Set<FieldName> metricNames
     Set<PhysicalTableDefinition> physDefs
     Set<TestApiDimensionName> dimNames
     Collection<Dimension> dims
+    Set<TableName> tableNames
+    Set<PhysicalTableDefinition> allDependentDefinitions
+
+    PhysicalTableDefinition definition1
+    PhysicalTableDefinition definition2
+    PhysicalTableDefinition definition3
+    PhysicalTableDefinition definition4
+    PhysicalTableDefinition definition5
+    PhysicalTableDefinition definition6
```

Comment:
Can we give these more descriptive names? Seeing it in code, `definition4` means nothing, but `selfDependentDefinition` is much more meaningful, for example.